editor/editor_main_screen.cpp

```diff
@@ -0,0 +1,298 @@
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "editor_main_screen.h"
+
+#include "core/io/config_file.h"
+#include "editor/plugins/editor_plugin.h"
+#include "editor_node.h"
+#include "editor_settings.h"
+#include "editor_string_names.h"
+#include "scene/gui/box_container.h"
+#include "scene/gui/button.h"
+
+void EditorMainScreen::_notification(int p_what) {
+	switch (p_what) {
+		case NOTIFICATION_READY: {
+			if (EDITOR_3D < buttons.size() && buttons[EDITOR_3D]->is_visible()) {
+				// If the 3D editor is enabled, use this as the default.
+				select(EDITOR_3D);
+				return;
+			}
+
+			// Switch to the first main screen plugin that is enabled. Usually this is
+			// 2D, but may be subsequent ones if 2D is disabled in the feature profile.
+			for (int i = 0; i < buttons.size(); i++) {
+				Button *editor_button = buttons[i];
+				if (editor_button->is_visible()) {
+					select(i);
+					return;
+				}
+			}
+
+			select(-1);
+		} break;
+		case NOTIFICATION_THEME_CHANGED: {
+			for (int i = 0; i < buttons.size(); i++) {
+				Button *tb = buttons[i];
+				EditorPlugin *p_editor = editor_table[i];
+				Ref<Texture2D> icon = p_editor->get_icon();
+
+				if (icon.is_valid()) {
+					tb->set_icon(icon);
+				} else if (has_theme_icon(p_editor->get_name(), EditorStringName(EditorIcons))) {
+					tb->set_icon(get_theme_icon(p_editor->get_name(), EditorStringName(EditorIcons)));
+				}
+			}
+		} break;
+	}
+}
+
+void EditorMainScreen::set_button_container(HBoxContainer *p_button_hb) {
+	button_hb = p_button_hb;
+}
+
+void EditorMainScreen::save_layout_to_config(Ref<ConfigFile> p_config_file, const String &p_section) const {
+	int selected_main_editor_idx = -1;
+	for (int i = 0; i < buttons.size(); i++) {
+		if (buttons[i]->is_pressed()) {
+			selected_main_editor_idx = i;
+			break;
+		}
+	}
+	if (selected_main_editor_idx != -1) {
+		p_config_file->set_value(p_section, "selected_main_editor_idx", selected_main_editor_idx);
+	} else {
+		p_config_file->set_value(p_section, "selected_main_editor_idx", Variant());
+	}
+}
+
+void EditorMainScreen::load_layout_from_config(Ref<ConfigFile> p_config_file, const String &p_section) {
+	if (p_config_file->has_section_key(p_section, "selected_main_editor_idx")) {
+		int selected_main_editor_idx = p_config_file->get_value(p_section, "selected_main_editor_idx");
```

Comment:
Could be simplified to
```suggestion
	int selected_main_editor_idx = p_config_file->get_value(p_section, "selected_main_editor_idx", -1);
```